### PR TITLE
add jobs for maistra-3.0 branch of maistra/istio-operator

### PR DESCRIPTION
**Pull request by dgn**
cc @yxun @jwendell

### CODE REVIEW

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      - name: GOCACHE
+      - name: GOBIN
+      - name: GOMODCACHE
+      from: src-test
+      resources:
+        requests:
+          cpu: "2"
+          memory: 2Gi
+      timeout: 20m0s
+    workflow: servicemesh-istio-e2e-aws
```
Comment (jwendell) [inline ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml:128]: is this needed? you probrably copied this from 2.4, but it's confusing... cc @yxun

File: core-services/image-mirroring/supplemental-ci-images/mapping_supplemental_ci_images_ci [hunk 2]
```diff
@@ -56,6 +56,7 @@ quay.io/maistra-dev/maistra-builder:2.1 registry.ci.openshift.org/ci/maistra-bui
 quay.io/maistra-dev/maistra-builder:2.2 registry.ci.openshift.org/ci/maistra-builder:2.2
 quay.io/maistra-dev/maistra-builder:2.3 registry.ci.openshift.org/ci/maistra-builder:2.3
 quay.io/maistra-dev/maistra-builder:2.4 registry.ci.openshift.org/ci/maistra-builder:2.4
+quay.io/maistra-dev/maistra-builder:3.0 registry.ci.openshift.org/ci/maistra-builder:3.0
```
[the remark this answers is not in the document]
Comment (yxun) [reply core-services/image-mirroring/supplemental-ci-images/mapping_supplemental_ci_images_ci:59]: once https://github.com/openshift/release/pull/37395 is merged. This file change will not be in this PR.

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml (hunk 1 above)
Comment (jwendell) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml:128]: I'm talking specifically about the `workflow: servicemesh-istio-e2e-aws` line. Is it needed?

Comment (yxun) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml:128]: yes only for the pushing operator image job. Our building operator image make target does not work in a regular step. So I need to let it run in a workflow.

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-istio-operator-maistra-3.0-maistra-istio-operator-lint-3-0

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml (hunk 1 above)
Comment (dgn) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml:128]: I did copy it. no idea what it's for

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      cli: latest
+      commands: |
+        export QUAY_PASS=$(cat /maistra-secrets/password)
+        oc rsh -n "${MAISTRA_NAMESPACE}" "${MAISTRA_SC_POD}" \
+          entrypoint \
+          sh -c \
+          "docker login -u='maistra-dev+prow' \
+          -p=${QUAY_PASS} quay.io; \
+          make push \
+          IMAGE=quay.io/maistra-dev/istio-ubi8-operator:PR-${PULL_NUMBER}"
```
[the remark this answers is not in the document]
Comment (dgn) [reply ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml:110]: Makes sense. *makes mental note to patch maistra-assist accordingly*

**Comment by dgn** [comment]
/pj-rehearse pull-ci-maistra-istio-operator-maistra-3.0-istio-operator-push-image-3-0

**Comment by dgn** [comment]
/pj-rehearse pull-ci-maistra-istio-operator-maistra-3.0-maistra-istio-operator-gencheck-3-0

**Review by jwendell — APPROVED** [review]

**Pull request merged**